cool_seq_tool/data_sources/feature_overlap.py

```diff
@@ -0,0 +1,180 @@
+
+        # Restrict to only feature of interest: CDS (which has gene info)
+        df = df[df["type"] == "CDS"].copy()
+
+        # Get name from the info field
+        df["info_name"] = df["info"].apply(
+            lambda info: re.findall("Name=([^;]+)", info)[0]
+        )
+
+        # Get gene from the info field
+        df["gene"] = df["info"].apply(lambda info: re.findall("gene=([^;]+)", info)[0])
+
+        # Get chromosome names without prefix and without suffix for alternate
+        # transcripts
+        df["chromosome"] = df["chromosome"].apply(
+            lambda chromosome: chromosome.strip("chr").split("_")[0]
+        )
+        df["chromosome"] = df["chromosome"].astype(str)
+
+        # Convert start and stop to ints
+        df["cds_start"] = df["cds_start"].astype(int)
+        df["cds_stop"] = df["cds_stop"].astype(int)
+
+        # Only retain certain columns
+        df = df[
+            ["type", "chromosome", "cds_start", "cds_stop", "info_name", "gene"]
+        ]
+
+        return df
+
+    def _get_chr_from_alt_ac(self, identifier: str) -> str:
+        """Get chromosome given genomic identifier
+
+        :param identifier: Genomic identifier on GRCh38 assembly
+        :raises FeatureOverlapError: If unable to find associated GRCh38 chromosome
+        :return: Chromosome. 1..22, X, Y. No 'chr' prefix.
+        """
+        aliases, error_msg = self.seqrepo_access.translate_identifier(
+            identifier, "GRCh38"
+        )
+
+        if error_msg:
+            raise FeatureOverlapError(str(error_msg))
+
+        if not aliases:
+            raise FeatureOverlapError(
+                f"Unable to find GRCh38 aliases for: {identifier}"
+            )
+
+        chr_pattern = r"^GRCh38:(?P<chromosome>X|Y|([1-9]|1[0-9]|2[0-2]))$"
+        for a in aliases:
+            chr_match = re.match(chr_pattern, a)
+            if chr_match:
+                break
+
+        if not chr_match:
+            raise FeatureOverlapError(
+                f"Unable to find GRCh38 chromosome for: {identifier}"
+            )
+
+        chr_groupdict = chr_match.groupdict()
+        return chr_groupdict["chromosome"]
+
+    def get_grch38_cds_overlap(
+        self,
+        start: int,
+        end: int,
+        chromosome: Optional[str] = None,
+        identifier: Optional[str] = None,
+        residue_mode: ResidueMode = ResidueMode.RESIDUE,
+    ) -> Optional[Dict]:
+        """Get feature overlap for GRCh38 genomic data
+
+        :param start: GRCh38 start position
+        :param end: GRCh38 end position
+        :param chromosome: Chromosome. 1..22, X, or Y. If not provided, must provide
+            `identifier`. If both `chromosome` and `identifier` are provided,
+            `chromosome` will be used.
+        :param identifier: Genomic identifier on GRCh38 assembly. If not provided,
+            must identifier `chromosome`. If both `chromosome` and `identifier` are
```

Comment:
Is there a word missing in "If not provided, must identifier `chromosome`"?

Reply:
Genome needs identified @austinant

Reply:
Nice catch. I accidentally replaced the wrong word. If not provided, must provide chromosome